### ISSUE TITLE
Interval should be able to provide reference point besides "starts"

### ISSUE THREAD

**Issue by djvs** [rossta/montrose#82]
Let's say I want a range with every 3 days.  I want to slice that interval up between Feb 1 and Mar 3.  But I want it to be every 3 days starting on Feb 2, counting before or after:

```
> Montrose.r(interval: 3, between: Date.parse("Feb 1 2017")..Date.parse("Mar 3 2017"), starts: Date.parse("Feb 2 2017"), every: :day).to_a
> ## OR ALSO...
> Montrose.r(interval: 3, starts: Date.parse("Jan 31 2017"), between: Date.parse("Feb 1 2017")..Date.parse("Mar 3 2017"), every: :day).to_a                    
=> [2017-02-01 00:00:00 -0500,
 2017-02-04 00:00:00 -0500,
 2017-02-07 00:00:00 -0500,
 2017-02-10 00:00:00 -0500,
 2017-02-13 00:00:00 -0500,
 2017-02-16 00:00:00 -0500,
 2017-02-19 00:00:00 -0500,
 2017-02-22 00:00:00 -0500,
 2017-02-25 00:00:00 -0500,
 2017-02-28 00:00:00 -0500]
```

I literally installed the library for this use case and then found out it wouldn't work.  Or will it?

**Comment by rossta** [comment]
So I understand your use case more clearly, what results are you expecting to get?

**Comment by djvs** [comment]
That the range is "affixed" on a reference date, but happens within a range
whose boundaries do not include any of its members.  E.g., every tenth day,
next Wednesday being included, between next Monday and October 10th.

**Comment by rossta** [comment]
Got it. So far `:between` has been considered shorthand for `:starts` and `:until`. The behavior you're looking for is not currently supported but it could be; sounds like a good feature request.

**Comment by djvs** [comment]
Well I'll share with you the work around that I implemented. What I did was
take the reference date, for inclusion in the interval and its two
boundaries, and for each of those, reason out the first or last included
date in the interval, and then just count forward or backward from there
for the members.  Just guessing re: the Montrose implementation, it'd
probably be a matter of finding the first, if any, member included in the
interval, and then substituting it for the start parameter?
